src/main/java/chungus/Parser.java

```diff
@@ -0,0 +1,129 @@
+package chungus;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class contains the logic needed to parse user entered commands and
+ * dispatch them to appropriate handlers.
+ */
+class Parser {
+    private static final Pattern DEADLINE_PATTERN = Pattern.compile("^deadline\\s+(.+)\\s+/by\\s+(.+)$");
+    private static final Pattern EVENT_PATTERN = Pattern.compile("^event\\s+(.+)\\s+/from\\s+(.+)\\s+/to\\s+(.+)$");
+
+    private static final DateTimeFormatter DATETIME_FMT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Parses a command, and selects a suitable handler. This is a pure function.
+     * The handler returned is not safe to be ran multiple times.
+     * 
+     * @param cmd The raw command to parse.
+     * @return An appropriate handler for the command.
+     * @throws ChungusException When the expected command format is wrong.
+     */
+    public static Handler parse(String cmd) {
```

Comment:
Very long method. Perhaps abstract away the parsing so that it is easier to maintain and add new cases. But nice use of switch cases!

Reply:
I think this is fine for now as the length comes from the number of cases. Within each case we only extract some values and pass it to the handler so it's not actually doing much. Thanks for the suggestion though!